snc.sh

```diff
@@ -223,26 +223,30 @@ create_pvs "${CRC_PV_DIR}" 30
 # https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusterversion.md#setting-objects-unmanaged
 ${OC} patch clusterversion version --type json -p "$(cat cvo_override.yaml)"
 
-# Get the pod name associated with cluster-monitoring-operator deployment
 cmo_pod=$(${OC} get pod -l app=cluster-monitoring-operator -o jsonpath="{.items[0].metadata.name}" -n openshift-monitoring)
-# Disable the deployment/replicaset/statefulset config for openshift-monitoring namespace
-${OC} scale --replicas=0 deployment --all -n openshift-monitoring
-# Wait till the cluster-monitoring-operator pod is deleted
+${OC} delete deployment cluster-monitoring-operator -n openshift-monitoring
+# Wait till the cluster-monitoring-operator pod is deleted before deleting other resources
 ${OC} wait --for=delete pod/$cmo_pod --timeout=60s -n openshift-monitoring
-# Disable the statefulset for openshift-monitoring namespace
-${OC} scale --replicas=0 statefulset --all -n openshift-monitoring
+${OC} delete deployment,statefulset,daemonset --all -n openshift-monitoring
 
 # Delete the pods which are there in Complete state
 ${OC} delete pods -l 'app in (installer, pruner)' -n openshift-kube-apiserver
 ${OC} delete pods -l 'app in (installer, pruner)' -n openshift-kube-scheduler
 ${OC} delete pods -l 'app in (installer, pruner)' -n openshift-kube-controller-manager
 
-# Disable the deployment/replicaset for openshift-machine-api and openshift-machine-config-operator
-${OC} scale --replicas=0 deployment --all -n openshift-machine-api
-${OC} scale --replicas=0 deployment --all -n openshift-machine-config-operator
+mao_pod=$(${OC} get pod -l k8s-app=machine-api-operator -o jsonpath="{.items[0].metadata.name}" -n openshift-machine-api)
+${OC} delete deployment machine-api-operator -n openshift-machine-api
+# Wait till the machine-api-operator pod is deleted before deleting other resources
+${OC} wait --for=delete pod/$mao_pod --timeout=60s -n openshift-machine-api
+${OC} delete statefulset,deployment,daemonset --all -n openshift-machine-api
 
-# Set replica to 0 for openshift-insights
-${OC} scale --replicas=0 deployment --all -n openshift-insights
+mco_pod=$(${OC} get pod -l k8s-app=machine-config-operator -o jsonpath="{.items[0].metadata.name}" -n openshift-machine-config-operator)
+${OC} delete deployment machine-config-operator -n openshift-machine-config-operator
+# Wait till the machine-config-operator pod is deleted before deleting other resources
+${OC} wait --for=delete pod/$mco_pod --timeout=60s -n openshift-machine-config-operator
+${OC} delete statefulset,deployment,daemonset --all -n openshift-machine-config-operator
+
+${OC} delete statefulset,deployment,daemonset --all -n openshift-insights
```

Comment:
no need to delete the operator first for openshift-insights?

Reply:
It is standalone operator and only have single operator deployment so no need to do it.

Reply:
Even if we don't need to do it now, isn't it more future-proof (and more consistent) to handle it in the same way as all other operators?

Reply:
sure, we can add this as part of new PR.

Reply:
Since you add this code as part of this PR, it would have made much more sense to add it to this one rather than changing it afterwards.